keygen/src/keygen.rs

```diff
@@ -224,7 +218,21 @@ fn main() -> Result<(), Box<dyn error::Error>> {
             let seed = Seed::new(&mnemonic, &passphrase);
             let keypair = keypair_from_seed(seed.as_bytes())?;
 
-            output_keypair(&keypair, &outfile, "new")?;
+            if !matches.is_present("no_outfile") {
+                let mut path = dirs::home_dir().expect("home directory");
+                let outfile = if matches.is_present("outfile") {
+                    matches.value_of("outfile").unwrap()
+                } else {
+                    path.extend(&[".config", "solana", "id.json"]);
+                    path.to_str().unwrap()
+                };
+
+                if outfile != "-" {
+                    check_for_overwrite(&outfile, &matches);
```

Comment:
This check should go before the prompts because it will exit if the file already exists. I think you could make `outfile` an option and check it before calling `output_keypair`

Reply:
🤦‍♂